server/lib/subscriptions.js

```diff
@@ -88,6 +88,13 @@ export async function processOrderWithSubscription(options, order) {
         transaction = await paymentsLib.processOrder(order);
         orderProcessedStatus = 'success';
       } catch (error) {
+        if (error.stripeResponse) {
+          if (error.stripeResponse.paymentIntent) {
```

Comment:
Ideally `subscriptions` should be agnostic about what Stripe is, it's the responsibility of `server/paymentProviders/stripe/*` to deal with Stripe errors